nemo/collections/asr/metrics/bleu.py

```diff
@@ -147,30 +134,48 @@ def update(
                 ``[Time, Batch]`` (if ``batch_dim_index == 1``). Required for MultiTaskDecoding.
             input_ids: an int torch.Tensor of shape ``[Batch, Time]`` (if ``batch_dim_index == 0``) or
                 ``[Time, Batch]`` (if ``batch_dim_index == 1``). Required for MultiTaskDecoding.
+            tokenizers: an optional sequence of strings of ``length == batch size``. Each element is passed as
+                the SacreBLEU tokenizer type for corresponding element in batch. If a sequence element is ``None``,
+                the initial tokenizer type from ``BLEU.__init__`` is used. If ``tokenizers == None`` then all elements
+                in batch are tokenized with initial tokenizer type.
         """
-        references = []
+        if tokenizers is not None:
+            assert (
+                len(tokenizers) == targets_lengths.shape[0]
+            ), f"BLEU metrics configured for multiple tokenizers, but got only '{len(tokenizers)}' tokenizer keywords for '{targets_lengths.shape[0]}' samples"
+
         with torch.no_grad():
+            # get predictions
+            hypotheses = self.decode(predictions, predictions_lengths, predictions_mask, input_ids, targets)
+
+            # prepare references
             tgt_lenths_cpu_tensor = targets_lengths.long().cpu()
             targets_cpu_tensor = targets.long().cpu()
+
             # check batch_dim_index is first dim
             if self.batch_dim_index != 0:
-                targets_cpu_tensor = move_dimension_to_the_front(targets_cpu_tensor, self.batch_dim_index)
+                targets_cpu_tensor = _move_dimension_to_the_front(targets_cpu_tensor, self.batch_dim_index)
+
             # iterate over batch
             for ind in range(targets_cpu_tensor.shape[0]):
                 tgt_len = tgt_lenths_cpu_tensor[ind].item()
                 target = targets_cpu_tensor[ind][:tgt_len].numpy().tolist()
                 reference = self.decoding.decode_tokens_to_str(target)
-                references.append(reference)
-            hypotheses = self.decode(predictions, predictions_lengths, predictions_mask, input_ids, targets)
 
-        if self.log_prediction:
-            logging.info("\n")
-            logging.info(f"reference:{references[0]}")
-            logging.info(f"predicted:{hypotheses[0]}")
-
-        super().update(
-            [h.text for h in hypotheses], [references]
-        )  # Note: [references] since BLEU allows multiple references.
+                self.preds_len, self.target_len = _bleu_score_update(
```

Comment:
Don;t you need to call super.__update__ here?

Reply:
Nah, since this is now using the full super code: https://github.com/Lightning-AI/torchmetrics/blob/4c42cf2c29f48f0024bbceb612672c20006a88ec/src/torchmetrics/text/sacre_bleu.py#L125